File: packages/transform/src/features/misc/MiscCreateIsCloneTransformer.ts
import { MiscIsCloneProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscCreateIsCloneTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "misc.createIsClone",
      write: MiscIsCloneProgrammer.write,
    });
}

File: packages/transform/src/features/misc/MiscCreateIsCloneTransformer.go
package misc

import internal "github.com/samchon/typia/packages/transform/src/internal"

var MiscCreateIsCloneTransformer = miscCreateIsCloneTransformerNamespace{}

type miscCreateIsCloneTransformerNamespace struct{}

func (miscCreateIsCloneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return "(input => JSON.parse(JSON.stringify(input)))"
}
